src/__tests__/fetcher.test.ts

```diff
@@ -44,4 +44,15 @@ describe('Fetcher Test', () => {
       );
     });
   });
+
+  describe('Request Fetcher Fetch Test', () => {
+    it('fetch with bad status code', async () => {
+      nock(baseURL).get('/').reply(404, 'error');
+
+      const fetcher = new Fetcher();
+      await expect(fetcher.fetch(baseURL)).rejects.toThrow(
+        'Failed to download'
+      );
```

Comment:
Instead of checking the error string, it's probably better if we check to see that the proper error type is being throw.
```suggestion
      await expect(fetcher.fetch(baseURL)).rejects.toThrow(
        DownloadHTTPError
      );
```